classes/sushi/IR.php

```diff
@@ -371,4 +378,254 @@ public function getReportItems(): array
+        }
+        if (in_array('Access_Method', $this->attributesToShow)) {
+            array_push($columnRow, 'Access_Method');
+        }
+
+        array_push($columnRow, 'Metric_Type', 'Reporting_Period_Total');
+
+        if ($this->granularity == 'Month') {
+            $period = $this->getMonthlyDatePeriod();
+            foreach ($period as $dt) {
+                array_push($columnRow, $dt->format('M-Y'));
+            }
+        }
+
+        return [$columnRow];
+    }
+
+    /** Get TSV report rows */
+    public function getTSVReportItems(): array
+    {
+        $results = $this->getQueryResults();
+
+        $resultsGroupedBySubmission = $resultRows = [];
+        $resultsGroupedBySubmission = $results->groupBy('submission_id');
+
+        foreach ($resultsGroupedBySubmission as $submissionId => $submissionResults) {
+            $results = collect($submissionResults);
+
+            // get total numbers for every metric type
+            $metricsTotal['Total_Item_Investigations'] = $results->pluck('metric_investigations')->sum();
+            $metricsTotal['Unique_Item_Investigations'] = $results->pluck('metric_investigations_unique')->sum();
+            $metricsTotal['Total_Item_Requests'] = $results->pluck('metric_requests')->sum();
+            $metricsTotal['Unique_Item_Requests'] = $results->pluck('metric_requests_unique')->sum();
+
+            // filter here by requested metric types
+            foreach ($this->metricTypes as $metricType) {
+                // if the total numbers for the given metric type > 0,
+                // construct the result row
+                if ($metricsTotal[$metricType] > 0) {
+                    $submission = Repo::submission()->get($submissionId);
+                    if (!$submission || !$submission->getOriginalPublication()) {
+                        break;
+                    }
+                    $currentPublication = $submission->getCurrentPublication();
+                    $submissionLocale = $submission->getData('locale');
+                    $datePublished = $submission->getOriginalPublication()->getData('datePublished');
+
+                    $resultRow = [
+                        $currentPublication->getLocalizedTitle($submissionLocale), // Item
+                        $this->context->getData('publisherInstitution'), // Publisher
+                        '', // Publisher ID
+                        $this->platformName, // Platform
+                    ];
+
+                    if (in_array('Authors', $this->attributesToShow)) {
+                        $authors = $currentPublication->getData('authors');
+                        $authorRowValue = '';
+                        foreach ($authors as $author) {
+                            $authorRowValue = $author->getFullName(true, false, $submissionLocale);
+                            $orcid = $author->getOrcid();
+                            if (isset($orcid) && !empty($orcid)) {
```

Comment:
I think `!empty($orcid)` should be all that's needed because if `$author->getOrcid()` returns nothing, `$orcid` will be set to `null` and the `!empty($orcid)` check should be enough.